sql/src/main/java/org/apache/druid/sql/calcite/schema/SegmentMetadataCache.java

```diff
@@ -403,6 +403,13 @@ public void start() throws InterruptedException
   @VisibleForTesting
   void refresh(final Set<SegmentId> segmentsToRefresh, final Set<String> dataSourcesToRebuild) throws IOException
   {
+    long startMillis = System.currentTimeMillis();
```

Comment:
don't recommend using `currentTimeMillis` in any new code. I'd suggest using `nanoTime` and then converting it to the needed duration

Reply:
Why do you recommend `nanoTime`, @rohangarg ?

Reply:
since that is always monotonically increasing in a JVM's lifetime, I don't think `currentTimeMillis` is, since it comes from the system time.

Reply:
I think the monotonicity depends on the NTP updates (on cloud instances), so not sure doing from same thread affects it. Also regarding cheapness, I think both calls are cheap and infrequent enough to not become a bottleneck for performance consideration. Further, the performance of each is also platform dependent I think.
I felt that for elapsed durations, probably something closer to a timer is better than using clock times.

Anyways, I'm ok with using any method.

Reply:
Thanks for the feedback. `nanoTime` seems like the better option for measuring elapsed duration.